models/spike_detector.cpp

```diff
@@ -80,19 +82,37 @@ nest::spike_detector::init_buffers_()
 void
 nest::spike_detector::calibrate()
 {
-  if ( !user_set_precise_times_
-    && kernel().event_delivery_manager.get_off_grid_communication() )
+  if ( kernel().event_delivery_manager.get_off_grid_communication() )
   {
-    device_.set_precise( true, 15 );
-
-    LOG( M_INFO,
-      "spike_detector::calibrate",
-      String::compose(
-           "Precise neuron models exist: the property precise_times "
-           "of the %1 with gid %2 has been set to true, precision has "
-           "been set to 15.",
-           get_name(),
-           get_gid() ) );
+    // If precise models exist, it is necessary to ensure that precise_times
+    // property
+    // is set to true and the precision should increase from the default value
+    // of 3.
+
+    // Default precision value for simulations involving precise models.
+    long precision = 15;
+
+    if ( user_set_precision_value != -1 )
+    {
+      // If precision is set to a value by the user, it must be used instead of
+      // the default.
+      precision = user_set_precision_value;
+    }
+
+    device_.set_precise( true, precision );
+
+    if ( user_set_precise_times_ != true )
```

Comment:
One should not compare booleans to `true` or `false`, just `if ( not user_set_precise_times_ )`.